src/components/button-loading.tsx

```diff
@@ -0,0 +1,34 @@
+import { Spinner } from "@chakra-ui/react";
+import { Button } from "@freecodecamp/ui";
+
+interface ButtonLoadingProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement | HTMLAnchorElement> {
+  children: React.ReactNode;
+  isPending: boolean;
+}
+
+export function ButtonLoading({
+  onClick,
+  children,
+  isPending,
+  ...rest
+}: ButtonLoadingProps) {
+  return (
+    <Button
+      block={true}
+      onClick={onClick}
+      disabled={isPending}
+      style={{
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        opacity: isPending ? 0.5 : 1,
```

Comment:
You should not need to change the opacity. 
When the button is disabled, it would automatically change color.
The text's position should not change ideally when the spinner is displayed, but it is ok for the first iteration.

Reply:
For the text position not changing, we should probably use Chakra's own button. However, getting the styling to match our ui component is not fun.